### ISSUE TITLE
v_xml_cdr caller_id_name/caller_id_number value end-user usability issues when setting them to outbound_caller_id_* values

### ISSUE THREAD

**Issue by kpabijanskas** [fusionpbx/fusionpbx#2240]
Hi,

As per these lines ( https://github.com/fusionpbx/fusionpbx/blob/master/app/xml_cdr/resources/classes/xml_cdr.php#L248-L254 ) that essentially pick off the caller_id_name/caller_id_value database fields during CDR import, I believe this can cause some, in some cases significant, usability issues.

Now, as per the first few lines, it will pull in effective_caller_id_name/number (unless it needs to fail over to the caller profile), which on most outbound routes will get set to:

`effective_caller_id_name=${outbound_caller_id_name}
 effective_caller_id_number=${outbound_caller_id_number}`

The problem that this creates is that the outbound caller id name/number will get put into the caller_id_name/caller_id_number fields inside the v_xml_cdr database, and the CDR Source page will end up displaying your outbound caller id values. On internal calls this will end up being the extension number, while on outgoing calls, going via an outbound route containing the two lines above, it will get set to the outbound_caller_id_name/number instead.

While this is in itself fine and allows you to see your actual end caller id on the CDR front page, this creates an issue when you have multiple extensions all sharing the same outbound caller id name and number. Without having access to the xml/json data and getting into the cdr details, it is impossible to tell which particular extension placed this call.

This does not matter too much if the end user has access to the cdr details, but as you know, freeswitch cdr details are quite verbose and contain way more than (non-technical) end users should ever see, and concern themselves with for that matter. If they have access to the cdr details page, they can view the extension that made the call, but if they do not, it is unfortunately not possible for them to know who in particular made the call.

Should it may be the case that the CDR page Source column, on outgoing calls, always displays the extension that actually made the call, so the real source can be tracked by just looking at the cdr page (rather than potentially ending up with a number of possibilities and no way to verify without accessing the details page for the call)?

**Comment by markjcrane** [comment]
The regular user doesn't get access to the CDR details by default. As you stated it is too much details to share with them.

**Comment by kpabijanskas** [comment]
Exactly, but if they need to see who placed the call (for example, a supervisor that has access to multiple extensions - setting their follow me/monitoring cdrs/etc.), would it not make more sense for them to be able to actually tell who placed the outbound call, if their outbound caller id name/number is the same? The CDRs are not as useful as they can be if you cant tell who actually placed the call. 

Same applies to cdr exports. These do not include the extra details (which they don't need to), so when it comes to exports, it is completely impossible to tell who placed that call in this same scenario

**Comment by kpabijanskas** [comment]
Actually, while working on  pull request #2280, I notice that this also breaks extension summaries for outbound calls in some scenarios. When calculating extension summaries, SQL ends up matching on the extension number, but it compares to v_xml_cdr.caller_id_number or v_xml_cdr.destination_number fields. On outgoing calls, the caller_id_number gets set to the outbound_caller_id_number in extension settings (if default fusionpbx outbound route dialplan is used). This means that it no longer matches the source extension and thus it does not get counted. You simply end up with the extension summary showing 0 outbound calls with 0 seconds elapsed time, even if that's not true. The only way to get those calls counted is then a number alias, but I do not believe that's the intended usage of number alias in the first place! 

As such I think xml cdr's should definitely retain the source extension number in the source column.

What do you think @markjcrane ?
